crates/cast/bin/cmd/run.rs

```diff
@@ -58,6 +58,10 @@ pub struct RunArgs {
     #[arg(long, short)]
     label: Vec<String>,
 
+    /// Etherscan API key.
+    #[arg(long)]
+    pub etherscan_api_key: Option<String>,
```

Comment:
we should use `EtherscanOpts` here similarly to other commands https://github.com/foundry-rs/foundry/blob/c4a31a624874ab36284fca4e48d2197e43a62fbe/crates/cast/bin/args.rs#L916

Reply:
should we then also allow setting of a different chain id? (and do not retrieve it from provider if set)

Reply:
don't think this would be useful, it's probably safe to always trust provider

using opts just allows us to reuse clap/figment config for etherscan_api_key

Reply:
was thinking would be helpful in situations like https://github.com/foundry-rs/foundry/issues/9255 so one could run cast run against anvil and decoding with etherscan API of choice, but agree it's safer to not add this

Reply:
hmm, not sure allowing this would really help. it works when anvil is in fork mode because we are able to match the contract addresses with actual etherscan contracts. if it's a local chain, then contract addresses wouldn't match the addresses etherscan can provide and we are forced to only rely on openchain API which might not contain data for all events we need

Reply:
good point, yeah, missed that :+1:

Reply:
yep, lgtm!